### PR TITLE
refactor(protocol-designer): migrate steplist to TS

### PR DESCRIPTION
**Pull request by shlokamin**
# Overview

Migrate steplist to TS

### CODE REVIEW

File: protocol-designer/src/components/lists/PDListItem.tsx [hunk 1]
```diff
@@ -3,12 +3,13 @@ import * as React from 'react'
 import cx from 'classnames'
 import styles from './styles.css'
 
-type Props = {
-  className: string | null | undefined
+interface Props {
+  className?: string | null
```
Comment (celsasser) [inline protocol-designer/src/components/lists/PDListItem.tsx:7]: I found this to be a problem as well. Unfortunately TS doesn't treat them the same.

File: protocol-designer/src/steplist/formLevel/getNextDefaultMagnetAction/__tests__/getNextDefaultModuleAction.test.ts [hunk 2]
```diff
@@ -31,7 +32,13 @@ describe('getNextDefaultMagnetAction', () => {
     ]
     testCases.forEach(({ testMsg, orderedStepIds, expected }) => {
       it(testMsg, () => {
-        const savedForms = {
+        const savedForms: {
```
Comment (celsasser) [inline protocol-designer/src/steplist/formLevel/getNextDefaultMagnetAction/__tests__/getNextDefaultModuleAction.test.ts:35]: Just wondering whether you can use `Record<StepIdType, FormData>` for form types

File: protocol-designer/src/steplist/formLevel/handleFormChange/test/moveLiquid.test.ts [hunk 3]
```diff
@@ -200,7 +209,7 @@ describe('path should update...', () => {
 })
 
 describe('disposal volume should update...', () => {
-  let form
+  let form: { path: string; aspirate_wells: string[]; dispense_wells: string[]; volume: string; pipette: string; disposalVolume_checkbox: boolean; disposalVolume_volume: string }
```
Comment (celsasser) [inline protocol-designer/src/steplist/formLevel/handleFormChange/test/moveLiquid.test.ts:212]: Wanna make this guy a type (I think I see his mirror image below)?

File: protocol-designer/src/steplist/formLevel/stepFormToArgs/thermocyclerFormToArgs.ts [hunk 4]
```diff
@@ -4,7 +4,7 @@ import {
   ThermocyclerStateStepArgs,
 } from '@opentrons/step-generation'
 import { PROFILE_STEP } from '../../../form-types'
-import { FormData, ProfileItem, ProfileStepItem } from '../../../form-types'
+import type { FormData, ProfileItem, ProfileStepItem } from '../../../form-types'
```
Comment (celsasser) [inline protocol-designer/src/steplist/formLevel/stepFormToArgs/thermocyclerFormToArgs.ts:7]: I didn't know this was supported in TS. I think globally removed some of them :(

**Review by celsasser — APPROVED** [review]
That's a whole lotta converting. Tiny thoughts but approving cause it looks good and the world will keep spinning if you ignore them :)

**Comment by celsasser** [comment]
You did it! You created a PR with exactly 50 files changed. Nice job!

File: protocol-designer/src/steplist/formLevel/stepFormToArgs/thermocyclerFormToArgs.ts (hunk 4 above)
Comment (shlokamin) [reply protocol-designer/src/steplist/formLevel/stepFormToArgs/thermocyclerFormToArgs.ts:7]: all good! we can call them out in the main PR and change them back as needed 😄

**Pull request merged**